armadillo/src/main/java/org/molgenis/armadillo/command/impl/CommandsImpl.java

```diff
@@ -73,12 +77,30 @@ public CommandsImpl(
   synchronized <T> CompletableFuture<T> schedule(ArmadilloCommandImpl<T> command) {
     final ArmadilloSession session = armadilloSession;
     lastCommand = command;
-    CompletableFuture<T> result =
-        supplyAsync(() -> session.execute(command::evaluate), executorService);
+    Supplier<T> execution = withCurrentSecurityContext(() -> session.execute(command::evaluate));
+    CompletableFuture<T> result = supplyAsync(execution, executorService);
     command.setExecution(result);
     return result;
   }
 
+  private <V> Supplier<V> withCurrentSecurityContext(Supplier<V> supplier) {
```

Comment:
This was a bit tricky. The armadilloService now does the security checks. but the security context is only available on the thread that does the scheduling. So here we decorate a supplier to run in the current security context.

Reply:
I did not know you could do this. What are Suppliers and why should you use it?

Reply:
Supplier<V> is the functional interface for a function that has no arguments and returns V

Reply:
They are used as a replacement for Callable by the CompletableFuture framework.

Reply:
Ah nice!